torch/fx/experimental/symbolic_shapes.py

```diff
@@ -232,6 +233,19 @@ def bool_(self):
 
 
 if HAS_SYMPY:
+    # NOTE [ SymPy eval and assumptions ]
+    # In eval, we only return values in cases where we always want to evaluate.
+    # In other cases, the result will just be FloorDiv(a, b), which needs to be
+    # evaluated later if necessary.
+    #
+    # We also define is_real=True and provide _eval_* methods to make the SymPy
+    # assumptions system aware of Python floordiv semantics. For instance, this
+    # ensures that correct assumptions are propagated when working with SymPy
+    # Symbols. Two integer Symbols should return an integer result.
+    #
+    # https://peps.python.org/pep-0238/#semantics-of-floor-division
+    # https://docs.sympy.org/latest/guides/assumptions.html#implementing-assumptions-handlers
+    # https://docs.sympy.org/latest/guides/custom-functions.html#best-practices-for-eval
```

Comment:
Rewrote the note with some context + new sympy links.

Reply:
This should probably also mention how we check for types. That is, we allow for None.